speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DatabaseConnectionManager.cpp

```diff
@@ -198,35 +195,47 @@ void AsyncCheckConnection(void* p_scrHandle, int p_dbStatusControl, tDatabaseSet
     *p_isConnecting = false;
 }
 
-/// @brief                    Checks if a connection can be established between speed dreams and the database
+/// @brief                    Checks if a connection can be established between speed dreams and the database with saved settings
 /// @param  p_scrHandle       The screen handle for writing on the screen
 /// @param  p_dbStatusControl The status control handle to write letters to the screen
-void CheckConnection(void* p_scrHandle, int p_dbStatusControl, bool* p_isConnecting)
+/// @param  p_isConnecting    Checks if a connection is already being made
+void CheckSavedConnection(void* p_scrHandle, int p_dbStatusControl, bool* p_isConnecting)
 {
     if (*p_isConnecting) return;
     *p_isConnecting = true;
     std::thread t(AsyncCheckConnection, p_scrHandle, p_dbStatusControl, s_dbSettings, p_isConnecting);
     t.detach();
 }
 
+/// @brief                    Checks if a connection can be established between speed dreams and the database with current settings
+/// @param  p_scrHandle       The screen handle for writing on the screen
+/// @param  p_dbStatusControl The status control handle to write letters to the screen
+/// @param  p_isConnecting    Checks if a connection is already being made
+void CheckCurrentConnection(void* p_scrHandle, int p_dbStatusControl, bool* p_isConnecting)
+{
+    if (*p_isConnecting) return;
+    *p_isConnecting = true;
+    std::thread t(AsyncCheckConnection, p_scrHandle, p_dbStatusControl, s_tempDbSettings, p_isConnecting);
+    t.detach();
+}
+
```

Comment:
The reason i did split them is because these functions are called outside the class where i have no access to s_tempDbSettings and s_DbSettings. I will make one function with an extra argument and just call those in the other functions.